File: app/src/api/product-catalog/data-transfer-objects/pad-reviews.ts
import { Rating } from './rating';
import { Review } from './review';

export interface PadReviews {
    rating: Rating;
    values: Review[];
    stats: ReviewStarStat[];
}

export interface ReviewStarStat {
    stars: number;
    count: number;
    percentage: number;
}

File: api/src/Application/ProductCatalog/Read/DataTransferObjects/PadReviews.cs
using System.Collections.Generic;

namespace DetailingArsenal.Application.ProductCatalog {
    public record PadReviews : IDataTransferObject {
        public RatingReadModel Rating { get; }
        public IEnumerable<ReviewReadModel> Values { get; }
        public IEnumerable<ReviewStarStat> Stats { get; }

        public PadReviews(RatingReadModel rating, IEnumerable<ReviewReadModel> values, IEnumerable<ReviewStarStat> stats) {
            Rating = rating;
            Values = values;
            Stats = stats;
        }
    }
}